lib/spack/spack/concretize.py

```diff
@@ -367,7 +369,7 @@ def concretize_compiler_flags(self, spec):
         Default specs set at the compiler level will still be added later.
         """
 
-        if not spec.architecture.platform_os:
+        if not (spec.architecture.platform_os and spec.architecture.target):
             # Although this usually means changed, this means awaiting other
             # changes
```

Comment:
Reword: We haven't changed, but other changes need to happen before we continue.  `return True` here to force concretization to keep running.